File: android/app/src/main/java/com/example/android/AddFriendPopupActivity.java
package com.example.android;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.view.MotionEvent;
import android.view.View;
import android.view.Window;
import android.widget.EditText;
import android.widget.TextView;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

public class AddFriendPopupActivity extends Activity {
    TextView txtText;
    EditText editText;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        //타이틀바 없애기
        requestWindowFeature(Window.FEATURE_NO_TITLE);
        setContentView(R.layout.popup_layout);

        //UI 객체생성
        txtText = (TextView)findViewById(R.id.txtText);
        editText = (EditText)findViewById(R.id.friendID);
        //데이터 가져오기
        //Intent intent = getIntent();
        //String data = intent.getStringExtra("data");
        //txtText.setText(data);
    }

    //확인 버튼 클릭
    public void mOnClose(View v){

        String friendID_str = editText.getText().toString();
        //데이터 전달하기
        //intent로 stringd을 전달해서 rankactivity에서 파이어베이스에 데이터를 넣거나 아니면 바로 여기서 넣거나.
        Intent intent = new Intent();
        intent.putExtra("result", friendID_str);
        setResult(RESULT_OK, intent);


        //액티비티(팝업) 닫기
        finish();
    }

    @Override
    public boolean onTouchEvent(MotionEvent event) {
        //바깥레이어 클릭시 안닫히게
        if(event.getAction()==MotionEvent.ACTION_OUTSIDE){
            return false;
        }
        return true;
    }

    @Override
    public void onBackPressed() {
        //안드로이드 백버튼 막기
        return;
    }
}

File: Model/androi/app/src/main/java/org/tensorflow/lite/examples/posenet/AddFriendPopupActivity.kt
package org.tensorflow.lite.examples.posenet

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.view.MotionEvent
import android.view.View
import android.view.Window
import android.widget.EditText
import android.widget.TextView

class AddFriendPopupActivity : Activity() {
    private lateinit var txtText : TextView
    private lateinit var editText : EditText
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        // Delete TitleBar
        requestWindowFeature(Window.FEATURE_NO_TITLE)
        setContentView(R.layout.popup_layout)

        // Instantiate UI instance
        txtText  = findViewById<View>(R.id.txtText) as TextView
        editText = findViewById<View>(R.id.friendID) as EditText
    }

    // On Click "OK"
    fun mOnClose(v: View?) {
        val intent = Intent()
        intent.putExtra("result", editText.text.toString())
        setResult(RESULT_OK, intent)
        // Close Popup Activity
        finish()
    }

    // Prevent click outside of popup
    override fun onTouchEvent(event: MotionEvent): Boolean {
        return event.action != MotionEvent.ACTION_OUTSIDE
    }

    // Prevent Back button
    override fun onBackPressed() {
        return
    }
}